spec/controllers/games_controller_spec.rb

```diff
@@ -68,6 +68,15 @@
       game.reload
       expect(game.current_turn).to eq(1)
     end
+
+      it "should update the free player with the current player" do
+        game = FactoryGirl.create(:game, white_player_id: 0)
+        player = FactoryGirl.create(:player, email: 'meow@meow.com', password: 'MONORAILCAT')
+        sign_in player
+        player.join_game!(game)
```

Comment:
Got it. Added a test that checks the other side of the logic, and refactored them a bit to make it more readable.